src/main/scala/org/clulab/wm/eidos/EidosSystem.scala

```diff
@@ -28,140 +24,153 @@ class EidosSystem(val components: EidosComponents) {
   // def this(x: Object) = this() // Dummy constructor crucial for Python integration
 
   protected val debug = true
-
-  // Grounding is the first PostProcessing step(s) and it is pre-configured in Eidos.  Other things can take
-  // the resulting AnnotatedDocument and post-process it further.  They are not yet integrated into Eidos.
-  val postProcessors: Seq[PostProcessing] = Seq(components.ontologyHandler)
+  protected def nestedArgumentExpander = new NestedArgumentExpander
 
   // ---------------------------------------------------------------------------------------------
   //                                 Annotation Methods
   // ---------------------------------------------------------------------------------------------
 
   def annotateDoc(doc: Document): Document = {
-    // It is assumed and not verified that the document _has_not_ already been annotated.
+    // It is assumed and not verified that the document has _not_ already been annotated.
     components.proc.annotate(doc)
     doc
   }
 
-  // Annotate the text using a Processor and then populate lexicon labels
+  // Annotate the text using a Processor and then populate lexicon labels.
   def annotate(text: String): Document = {
-    val tokenized = components.proc.mkDocument(text, keepText = true) // Formerly keepText, must now be true
-    val annotated = annotateDoc(tokenized)
+    val tokenizedDoc = components.proc.mkDocument(text, keepText = true) // Formerly keepText, must now be true
+    val annotatedDoc = annotateDoc(tokenizedDoc)
 
-    annotated
+    annotatedDoc
   }
 
   // ---------------------------------------------------------------------------------------------
   //                                 Extraction Methods
   // ---------------------------------------------------------------------------------------------
 
-  def extractFrom(doc: Document): Vector[Mention] = {
+  def extractMentionsFrom(doc: Document): Seq[Mention] = {
 
-    def extractEventsFrom(doc: Document, state: State): Vector[Mention] = {
-      val extractedEvents = components.engine.extractFrom(doc, state).toVector
-      val mostCompleteEvents = components.actions.keepMostCompleteEvents(extractedEvents, State(extractedEvents)).toVector
+    def extractMentions(state: State): Seq[Mention] = {
```

Comment:
I think the Vectors were vectors bc the Reach team got bit by something lazy? Or something? I wasn't involved, but food for thought... you can check with marco if you want more info

Reply:
Thank you for all the invaluable comments!  I will check on it.

Reply:
I sent the message.  I would think that at most only the first toVector is needed to "materialize" any lazy stuff.